7thAce/SC2CCM
# Added logging, removed binary files from git Repository

## Related Issues:
* https://github.com/7thAce/SC2CCM/issues/2
* https://github.com/7thAce/SC2CCM/issues/5

## Description

Added logging with log rotation to the mod manager (Update has not been changed in this PR). Logs are currently saved in AppData, next to the config file. Logs are also printed out to the console for dev purposes.

Added a .gitignore based on [Microsoft's .NET template](https://github.com/github/gitignore/blob/main/VisualStudio.gitignore). This ignores a lot of binary files (e.g. intermediate compilation files) and user-specific files (e.g. .suo). By ignoring binary compilation files and user specific files, the repository will be kept small, there will be less merge conflicts, and there should be less issues for contributors and their IDEs. The .gitignore template also accounts for several different types of IDEs, such as Visual Studio, Rider, and Visual Studio Code.

## Test Plan

* Build and run the "StarCraft Mod Manager" project
* Open it up and do a few actions
* Open up "%appdata%/SC2CCM/mod_manager.log", you should see a log file describing some of the actions which happened
* Close and reopen the app
* Open up %appdata%/SC2CCM/mod_manager.log.old, it should be your previous log file
* Open up %appdata%/SC2CCM/mod_manager.log, it should be different
